File: enode/src/main/java/com/enodeframework/domain/IAggregateRootInternalHandlerProvider.java
package com.enodeframework.domain;

import com.enodeframework.common.function.Action2;
import com.enodeframework.eventing.IDomainEvent;

/**
 * Defines a provider interface to provide the aggregate root internal handler.
 */
public interface IAggregateRootInternalHandlerProvider {

    /**
     * Get the internal event handler within the aggregate.
     *
     * @param aggregateRootType
     * @param anEventType
     * @return
     */
    Action2<IAggregateRoot, IDomainEvent> getInternalEventHandler(Class<? extends IAggregateRoot> aggregateRootType, Class<? extends IDomainEvent> anEventType);
}

File: enode/src/main/java/org/enodeframework/domain/IAggregateRootInternalHandlerProvider.kt
package org.enodeframework.domain

import org.enodeframework.common.function.Action2
import org.enodeframework.eventing.IDomainEvent
import java.util.*

/**
 * Defines a provider interface to provide the aggregate root internal handler.
 */
interface IAggregateRootInternalHandlerProvider {
    /**
     * Get the internal event handler within the aggregate.
     */
    fun getInternalEventHandler(aggregateRootType: Class<out IAggregateRoot>, eventType: Class<out IDomainEvent<*>>): Action2<IAggregateRoot, IDomainEvent<*>>
}